docs/blog-post.md

```diff
@@ -0,0 +1,286 @@
+
+```scala
+case class Message(animal: Animal) extends MultiDocPrintService
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(
+  Array(
+    new JsonSubTypes.Type(value = classOf[Lion], name = "lion"),
+    new JsonSubTypes.Type(value = classOf[Tiger], name = "tiger")))
+sealed trait Animal
+
+final case class Lion(name: String) extends Animal
+final case class Tiger(name: String) extends Animal
+
+```
+
+#### Example 2
+
+If a Scala object is defined:
+```scala
+case object Tick
+```
+
+There **will not be exceptions** during serialization but Jackson will create
+another instance of `Tick` instead of restoring the `object Tick`'s underlying singleton. So, deserialization will end up very bad...
+
+```scala
+actorRef ! Tick
+
+// Inside the actor:
+def receive = {
+  case Tick => // this won't get matched !!
+} // message will be unhandled !!
+```
+
+#### But there is a better way to do serialization!
+To get rid of such problems, our circe-based Akka serializer can be used. [Circe Akka Serializer](https://github.com/VirtusLab/akka-serialization-helper/tree/main/circe-akka-serializer) (part of the `akka-serialization-helper` toolbox) uses Circe codecs, derived using [Shapeless](https://circe.github.io/circe/codecs/auto-derivation.html),
+that are generated during compile time (so serializer won't crash during runtime as reflection-based serializers may do).
+
+Circe Akka Serializer is really easy to use, just add the following to project dependencies...
+
+```scala
+import org.virtuslab.ash.AkkaSerializationHelperPlugin
+
+lazy val app = (project in file("app"))
+  // ...
+  .settings(libraryDependencies += AkkaSerializationHelperPlugin.circeAkkaSerializer)
+```
+
+and create a custom serializer by extending `CirceAkkaSerializer` base class:
+```scala
+import org.virtuslab.ash.circe.CirceAkkaSerializer
+
+class ExampleSerializer(actorSystem: ExtendedActorSystem)
+    extends CirceAkkaSerializer[MySerializable](actorSystem) {
+
+  override def identifier: Int = 41
+
+  override lazy val codecs = Seq(Register[CommandOne], Register[CommandTwo])
+
+  override lazy val manifestMigrations = Nil
+
+  override lazy val packagePrefix = "org.project"
+}
+```
+and that's it! You have safe and sound circe-based serializer to cope with serialization of your objects.
+
+### 4. Missing Codec registration
+
+Last thing that might cause unexpected issues during serialization goes here: if a codec is not registered, a runtime exception will occur.
+```scala
+import org.virtuslab.ash.circe.CirceAkkaSerializer
+import org.virtuslab.ash.circe.Register
+
+class ExampleSerializer(actorSystem: ExtendedActorSystem)
+  extends CirceAkkaSerializer[MySerializable](actorSystem) {
+  // ...
+  override lazy val codecs = Seq(Register[CommandOne]) // WHOOPS someone forgot to register CommandTwo...
+}
+```
+```shell
```

Comment:
Nit: turn off syntax coloring here, `shell` coloring gives some random effects like `for` treated as keyword

Reply:
I just changed this one from ```shell to ``` (without shell) - as this code block is not a command, but just an output with Exception message.